build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1288,6 +1608,218 @@ message WaitExecutionRequest {
+  // omitted.
+  string instance_name = 1;
+
+  // The hash of an [Action][build.bazel.remote.execution.v2.Action] that acts as a weak fingerprint
+  // key to look up potential action results that must be compared to the client's current filesystem
+  // contents to get a match. On a match, the corresponding build outputs in the form of an
+  // [ActionResult][build.bazel.remote.execution.v2.ActionResult] can be retrieved from
+  // [ActionCache.GetActionResult][build.bazel.remote.execution.v2.ActionCache.GetActionResult].
+  //
+  // In two-phase caching an `Action` contains a normal reference to a
+  // [Command][build.bazel.remote.execution.v2.Command] but the `input_root_digest`
+  // can refer to a subset of the files that are predicted to be needed to run the `Action`.
+  // The input root can contain all files needed for the command, in which case the weak
+  // fingerprint should lead to just one output (or none if there are no cache hits).
+  // When the input root contains fewer files than those actually needed by the action,
+  // the weak fingerprint gets less specific, and the number of possible matches grows.
+  // It's possible for the input root to contain zero files, which is a very weak
+  // fingerprint as it can match any output that matches the `Command`.
+  // Less-specific weak fingerprints lead to more I/O and time spent matching pathsets against
+  // the client filesystem; adding in any predictions about the filesystem "focuses"
+  // the lookup to more-sepcific results.
+  Digest weak_fingerprint_action_digest = 2;
+
+  // A hint to the server to request inlining pathsets in the
+  // [GetPotentialActionResultsResponse][build.bazel.remote.execution.v2.GetPotentialActionResultsResponse]
+  // message.
+  bool inline_pathsets = 3;
+
+  // A maximum page size to request. If present, the server will respond with no more
+  // than this many items. Regardless of whether a page size is specified, the
+  // server may place its own limit on the number of items to be returned and
+  // require the client to retrieve more items using a subsequent request.
+  int32 page_size = 4;
+
+  // A page token, which must be a value received in a previous
+  // [GetPotentialActionResultsResponse][build.bazel.remote.execution.v2.GetPotentialActionResultsResponse].
+  // If present, the server will use that token as an offset, returning only
+  // that page and the ones that succeed it.
+  string page_token = 5;
+}
+
+// A GetPotentialActionResultsResponse returns a set of zero or more (pathset, strong fingerprint)
+// results from a two-phase cache lookup call to
+// [ActionCache.GetPotentialActionResults][build.bazel.remote.execution.v2.ActionCache.GetPotentialActionResults].
+message GetPotentialActionResultsResponse {
+  // A set of possible matches to the client filesystem along with a reference to build outputs.
+  repeated ObservedPathSetAndStrongFingerprint potential_action_matches = 1;
```

Comment:
>ObservedPathSetAndStrongFingerprint [](start = 11, length = 35)

I think this should be `ObservedPathSetDigestAndStorngFingerprint`. We don't want to fetch/deserialize all path sets from CAS. If the one that matches the filesystem is found, then we should use it. Another advanced scenario is to have augmented weak fingerprint.

Reply:
Ah, ignore this. Down there, the message actually contains the digest of observed path set.

---
In reply to: [459020941](https://github.com/erikma/remote-apis/pull/1#discussion_r459020941) [](ancestors = 459020941)